ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,277 @@
+        system(['wb_command -cifti-smoothing ' vn_file_path ' 1.47106851007471610247 1.47106851007471610247 COLUMN ' vn_smooth_file_path ' -left-surface ' fs_path '/' subj '.L.midthickness_MSMAll.32k_fs_LR.surf.gii -right-surface ' fs_path '/' subj '.R.midthickness_MSMAll.32k_fs_LR.surf.gii']);
+
+        CIFTIMSMAll=ciftiopen(CIFTIMSMAll_file,'wb_command');
+        CIFTIMSMAllVN=ciftiopen(vn_file_path,'wb_command');
+        CIFTIMSMAll.cdata=CIFTIMSMAll.cdata./repmat(CIFTIMSMAllVN.cdata,1,size(CIFTIMSMAll.cdata,2));           
+        CIFTIMSMAllSmooth=ciftiopen([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_MSMAll_' num2str(FinalSpatialSmoothingFWHM) '.dscalar.nii'],'wb_command');
+        CIFTIMSMAllSmoothVN=ciftiopen([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_Atlas_MSMAll_hp' hp '_clean_vn_' num2str(FinalSpatialSmoothingFWHM) '.dscalar.nii'],'wb_command');
+        CIFTIMSMAllSmooth.cdata=CIFTIMSMAllSmooth.cdata./repmat(CIFTIMSMAllSmoothVN.cdata,1,size(CIFTIMSMAllSmooth.cdata,2));           
+        %CIFTIDropouts=ciftiopen([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.dscalar.nii'],'wb_command');
+        cifti_dropout_file=[SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.dscalar.nii'];
+
+        CIFTIDropouts=ciftiopen(cifti_dropout_file,'wb_command');
+        WMPARC=read_avw([SubjFolderlist '/MNINonLinear/ROIs/wmparc.' Resolution '.nii.gz']);
+        ROIFolder=[SubjFolderlist '/MNINonLinear/ROIs'];
+
+        system(['wb_command -volume-label-import ' ROIFolder '/wmparc.' Resolution '.nii.gz ' CSFLabelFile ' ' ROIFolder '/CSFReg.' Resolution '.nii.gz -discard-others -drop-unused-labels']);
+        system(['wb_command -volume-label-import ' ROIFolder '/wmparc.' Resolution '.nii.gz ' WMLabelFile ' ' ROIFolder '/WMReg.' Resolution '.nii.gz -discard-others -drop-unused-labels']);
+
+        WM=read_avw([SubjFolderlist '/MNINonLinear/ROIs/WMReg.' Resolution '.nii.gz']);
+        CSF=read_avw([SubjFolderlist '/MNINonLinear/ROIs/CSFReg.' Resolution '.nii.gz']);
+        % dropout file is not ready for lifespan
+        %DROPOUT=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.nii.gz']);
+        dropout_file=[SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.nii.gz'];
+
+        DROPOUT=read_avw(dropout_file);
+        VOLUME=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC.nii.gz']);
+        VOLUME=reshape(VOLUME,size(VOLUME,1)*size(VOLUME,2)*size(VOLUME,3),size(VOLUME,4));
+        Volume.cdata=VOLUME(std(VOLUME,[],2)>0,:);
+        WMPARC=reshape(WMPARC,size(WMPARC,1)*size(WMPARC,2)*size(WMPARC,3),size(WMPARC,4));
+        wmparc.cdata=WMPARC(std(VOLUME,[],2)>0,:);
```

Comment:
Please don't use `.cdata` to pretend something is a cifti structure, it is confusing.